File: backend/src/main/java/com/legaltrack/service/TrackedCaseService.java
package com.legaltrack.service;

import com.legaltrack.dto.common.PagedResponse;
import com.legaltrack.dto.tracked.TrackCaseRequest;
import com.legaltrack.dto.tracked.TrackedCaseDto;
import com.legaltrack.dto.tracked.UpdateTrackedCaseRequest;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface TrackedCaseService {
    TrackedCaseDto trackCase(TrackCaseRequest request);
    PagedResponse<TrackedCaseDto> getMyTrackedCases(String query, Pageable pageable);
    List<TrackedCaseDto> getAllMyTrackedCases();
    TrackedCaseDto updateTrackedCase(Long id, UpdateTrackedCaseRequest request);
    void stopTracking(Long id);
    void toggleNotifications(Long id, boolean enabled);
}

File: frontend/src/services/trackedCaseService.js
import api from './api';

export const trackedCaseService = {
  getMyTrackedCases: async (query = '', page = 0, size = 10) => {
    const res = await api.get(`/tracked-cases?query=${encodeURIComponent(query)}&page=${page}&size=${size}`);
    return res.data;
  },

  getAllTrackedCases: async () => {
    const res = await api.get('/tracked-cases/all');
    return res.data;
  },

  trackCase: async (data) => {
    const res = await api.post('/tracked-cases', data);
    return res.data;
  },

  updateTrackedCase: async (id, data) => {
    const res = await api.put(`/tracked-cases/${id}`, data);
    return res.data;
  },

  stopTracking: async (id) => {
    const res = await api.delete(`/tracked-cases/${id}`);
    return res.data;
  },

  toggleNotifications: async (id, enabled) => {
    const res = await api.patch(`/tracked-cases/${id}/notifications?enabled=${enabled}`);
    return res.data;
  }
};
